pkg/daemon/drain_test.go

```diff
@@ -0,0 +1,343 @@
+			oldConfig:      machineConfigs["mc1"],
+			newConfig:      machineConfigs["mc1"],
+			expectedAction: false,
+		},
+		{
+			// perform drain: reboot action is present
+			actions:        []string{postConfigChangeActionNone, postConfigChangeActionReboot},
+			oldConfig:      machineConfigs["mc1"],
+			newConfig:      machineConfigs["mc1"],
+			expectedAction: true,
+		},
+		// below tests are run when only crio reload action is present
+		{
+			// skip drain: no changes in registry config
+			actions:        []string{postConfigChangeActionReloadCrio},
+			oldConfig:      machineConfigs["mc1"],
+			newConfig:      machineConfigs["mc1"],
+			expectedAction: false,
+		},
+		{
+			// skip drain: only new registry added with mirror-by-digest-only set to true
+			actions:        []string{postConfigChangeActionReloadCrio},
+			oldConfig:      machineConfigs["mc3"],
+			newConfig:      machineConfigs["mc1"],
+			expectedAction: false,
+		},
+		{
+			// perform drain: only new registry added with mirror-by-digest-only set to false
+			actions:        []string{postConfigChangeActionReloadCrio},
+			oldConfig:      machineConfigs["mc5"],
+			newConfig:      machineConfigs["mc6"],
+			expectedAction: true,
+		},
+		{
+			// perform drain: one or more registry has been removed
+			actions:        []string{postConfigChangeActionReloadCrio},
+			oldConfig:      machineConfigs["mc1"],
+			newConfig:      machineConfigs["mc3"],
+			expectedAction: true,
+		},
+		{
+			// skip drain: only new mirrors got added to the registry with mirror-by-digest-only set to true for registry
+			actions:        []string{postConfigChangeActionReloadCrio},
+			oldConfig:      machineConfigs["mc1"],
+			newConfig:      machineConfigs["mc2"],
+			expectedAction: false,
+		},
+		{
+			// perform drain: only new mirrors got added to the registry with mirror-by-digest-only set to false for registry
+			actions:        []string{postConfigChangeActionReloadCrio},
+			oldConfig:      machineConfigs["mc6"],
+			newConfig:      machineConfigs["mc7"],
+			expectedAction: true,
+		},
+		{
+			// perform drain: one or more mirror has been removed from a registry
+			actions:        []string{postConfigChangeActionReloadCrio},
+			oldConfig:      machineConfigs["mc2"],
+			newConfig:      machineConfigs["mc1"],
+			expectedAction: true,
+		},
+		{
+			// perform drain: either item from unqualified-search-registries has been removed or ordering has been changed
+			actions:        []string{postConfigChangeActionReloadCrio},
+			oldConfig:      machineConfigs["mc4"],
+			newConfig:      machineConfigs["mc3"],
+			expectedAction: true,
+		},
+		{
+			// skip drain: only additional unqualified-search-registries has been added
+			actions:        []string{postConfigChangeActionReloadCrio},
+			oldConfig:      machineConfigs["mc3"],
+			newConfig:      machineConfigs["mc4"],
+			expectedAction: false,
+		},
+		{
+			// perform drain: prefix value of one or more registry has changed
```

Comment:
Very minor, comparing 7 to 1, I don't think `prefix` value was changed, instead a registries.mirror was added with mirror-by-digest-only true (this is fine) but the other registry's mirror-by-digest-only changed from true to false, which is an unsafe change covered by the last test in the list. Did you mean to use other MCs here?

Reply:
good catch Jerry! fixed it by adding a new MC that has only prefix change.